lib/puppetserver/ca/action/generate.rb

```diff
@@ -143,15 +140,25 @@ def generate_pki(settings, signing_digest, subject_alt_names = '')
             [settings[:cakey], int_key],
           ]
 
+          errors = FileSystem.check_for_existing_files(public_files.map { |f| f.first })
+          errors += FileSystem.check_for_existing_files(private_files.map { |f| f.first })
+
+          if errors.any?
+            errors << "If you would really like to replace your CA, please delete the existing files first.
```

Comment:
This is the preferred way to do it in clojure, but for some reason I thought HEREDOCs were the preferred way to do multiline strings with newlines in them in Ruby. I can't find any consensus on the Internet or even recommendations in Rubocop. So I'm probably wayy off on that. Do you know if there's any preference within the company?

Reply:
I think a HEREDOC would be more  idiomatic. I'll update.